tinygrad/runtime/graph/hcq.py

```diff
@@ -11,6 +11,7 @@ class HCQGraph(MultiGraphRunner):
   def __init__(self, jit_cache: List[ExecItem], input_rawbuffers: List[Buffer], var_vals: Dict[Variable, int]):
     super().__init__(jit_cache, input_rawbuffers, var_vals)
     self.devices = list(set(cast(HCQCompiled, d) for ji in jit_cache for d in [Device[cast(Buffer, x).device] for x in ji.bufs]))
+    self.bufs_stride, self.vars_stride = (2, 4) if self.devices[0].dname.startswith('QCOM') else (1, 1)
```

Comment:
What is this and why does qcom need it?

Reply:
This is a hack to support hcq. amd/nv populates its bufs and vars sequentially 0x140, 0x148, ....
qcom does it differently, 0x140, 0x150 and so on. thus, aligning them by 16 bytes.
see hexdump of bufs in qcom constants:
```
00000140: 00 B0 BA FF 07 00 00 00  00 00 00 00 00 00 00 00  ................
00000150: 00 00 BD FF 07 00 00 00  00 00 00 00 00 00 00 00  ................
```
Of course, it's not the best decision to set these strides here, but hcq api currently does not support exporting this functionality to the device implementation. I think I can come up with something better in a separate PR after this is merged. Or do it just here, if this is too "hacky".

Reply:
master have an abstraction for this now